kratos/input_output/vtk_output.cpp

```diff
@@ -912,20 +912,23 @@ void VtkOutput::WriteIntegrationVectorContainerVariable(
         return;
     }
 
-    const int res_size = static_cast<int>((rContainer.begin()->GetValue(rVariable)).size());
+    // determining size of results
+    const auto& r_process_info = mrModelPart.GetProcessInfo();
+    std::vector<TVarType> tmp_result;
+    rContainer.begin()->CalculateOnIntegrationPoints(rVariable, tmp_result, r_process_info);
+    const int res_size = tmp_result[0].size();
 
     rFileStream << rVariable.Name() << " " << res_size << " " << rContainer.size() << "  float\n";
 
     // Auxiliar values
-    const auto& r_process_info = mrModelPart.GetProcessInfo();
     auto& r_this_geometry_begin = (rContainer.begin())->GetGeometry();
     const GeometryData::IntegrationMethod this_integration_method = (rContainer.begin())->GetIntegrationMethod();
     const auto& r_integration_points = r_this_geometry_begin.IntegrationPoints(this_integration_method);
     const SizeType integration_points_number = r_integration_points.size();
 
     TVarType aux_value;
     for (auto& r_entity : rContainer) { // TODO: CalculateOnIntegrationPoints should be const methods
-        aux_value = rVariable.Zero();
+        aux_value = ZeroVector(res_size);
```

Comment:
for the record, `rVariable.Zero()` does not work for variables of type `Vector`, since those have a size of zero by default